Extend Merge Capabilities (#739)
databricks/dbt-databricks: CHANGELOG.md

- Add support for serverless job clusters on python models ([706](https://github.com/databricks/dbt-databricks/pull/706))
- Add 'user_folder_for_python' config to switch writing python model notebooks to the user's folder ([706](https://github.com/databricks/dbt-databricks/pull/706))

> - Merge capabilities are extended ([739](https://github.com/databricks/dbt-databricks/pull/739)) to include the support for the follwing features:

Reviewer: Can you add a thank you call out (search 'thanks for examples) to give yourself credit for this work :)?

Author: ```suggestion
- Merge capabilities are extended ([739](https://github.com/databricks/dbt-databricks/pull/739)) to include the support for the following features (thanks @mi-volodin):
```

  - `with schema evolution` clause (requires Databricks Runtime 15.2 or above);

>   - `when not matched by source` clause, only for `delete` action

Author: ```suggestion
  - `when not matched by source` clause, only for `delete` action;
```

>   - `when not matched on source` clause, only for `delete` action
>   - `matched`, `not matched` and `not matched on source` condition clauses;

Author: ```suggestion
  - `when not matched by source` clause, only for `delete` action
  - `matched`, `not matched` and `not matched by source` condition clauses;
```
